apps/ios/Shared/Model/SimpleXAPI.swift

```diff
@@ -1521,19 +1586,23 @@ func active(_ user: User) -> Bool {
     user.id == ChatModel.shared.currentUser?.id
 }
 
-func chatItemSimpleUpdate(_ user: User, _ aChatItem: AChatItem) {
+func chatItemSimpleUpdate(_ user: User, _ aChatItem: AChatItem) async {
     let m = ChatModel.shared
     let cInfo = aChatItem.chatInfo
     let cItem = aChatItem.chatItem
-    if active(user) && m.upsertChatItem(cInfo, cItem) {
-        NtfManager.shared.notifyMessageReceived(user, cInfo, cItem)
+    if active(user) {
+        if await MainActor.run(body: { m.upsertChatItem(cInfo, cItem) }) {
+            NtfManager.shared.notifyMessageReceived(user, cInfo, cItem)
+        }
     }
 }
 
-func updateContactsStatus(_ contactRefs: [ContactRef], status: NetworkStatus) {
+func updateContactsStatus(_ contactRefs: [ContactRef], status: NetworkStatus) async {
     let m = ChatModel.shared
-    for c in contactRefs {
-        m.networkStatuses[c.agentConnId] = status
+    await MainActor.run {
```

Comment:
it could be more efficient to build a map outside of MainActor.run and then merge networkStatuses with it